File: .semgrep/tests/naming/typescript/ambiguous_suffix_test.ts
// ruleid: naming.ambiguous-suffix-class-typescript
class UserManager {
    users: string[] = [];
}

// ruleid: naming.ambiguous-suffix-class-typescript
class TaskEngine {
    tasks: string[] = [];
}

// ruleid: naming.ambiguous-suffix-interface-typescript
interface ConnectionService {
    connect(): void;
}

// ruleid: naming.ambiguous-suffix-type-alias-typescript
type AppRuntime = {
    name: string;
};

// ok: naming.ambiguous-suffix-class-typescript
class UserRegistry {
    users: string[] = [];
}

// ok: naming.ambiguous-suffix-interface-typescript
interface DataStore {
    get(): string;
}

// ok: naming.ambiguous-suffix-type-alias-typescript
type TaskExecutor = {
    run(): void;
};

File: .semgrep/tests/naming/go/ambiguous_suffix_test.go
package main

// ruleid: naming.ambiguous-suffix-struct-go
type UserManager struct {
	users []string
}

// ruleid: naming.ambiguous-suffix-struct-go
type TaskEngine struct {
	tasks []string
}

// ruleid: naming.ambiguous-suffix-struct-go
type ConnectionUtil struct{}

// ruleid: naming.ambiguous-suffix-struct-go
type UserFacade struct{}

// ruleid: naming.ambiguous-suffix-struct-go
type AuthService struct{}

// ruleid: naming.ambiguous-suffix-struct-go
type AppRuntime struct{}

// ok: naming.ambiguous-suffix-struct-go
type UserRegistry struct{}

// ok: naming.ambiguous-suffix-struct-go
type TaskScheduler struct{}

// ok: naming.ambiguous-suffix-struct-go
type AuthPolicy struct{}

// ok: naming.ambiguous-suffix-struct-go
type OrderFactory struct{}

// ruleid: naming.ambiguous-suffix-interface-go
type DataManager interface {
	Get() string
}

// ok: naming.ambiguous-suffix-interface-go
type DataStore interface {
	Get() string
}
